### PR TITLE
Fix deadlock

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: proxy/frontend_handler.go [hunk 1]
```diff
@@ -41,28 +42,29 @@ func (h *FrontendHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
...
 			switch message.Type {
 			case common.Body:
-				ws.WriteMessage(1, []byte(message.Body))
+				ws.SetWriteDeadline(time.Now().Add(10 * time.Second))
+				if err := ws.WriteMessage(1, []byte(message.Body)); err != nil {
+					h.closeConnection(ws)
+				}
 			case common.Close:
 				h.closeConnection(ws)
-				return
```
Comment (ibuildthecloud) [inline proxy/frontend_handler.go:58]: Why did you remove this return?

Comment (cjellick) [inline proxy/frontend_handler.go:58]: This was the critical line causing the bug. This goroutine woiuld exit and then there would be nothing to read from the channel. So now, the only thing that can exit this goroutine is the channel exiting. And the channel is closed via `defer h.backend.closeConnection(hostKey, msgKey)` above

**Comment by ibuildthecloud** [comment]
LGTM, good snag.
